[VDG] MusicBox - Fix confusing messages

soosr opened the pull request:
Co-worked with @nopara73.

We will be aiming to have the fewest messages in the default MusicBox workflow while having as many clear and straightforward messages as we need for the edge cases.

soosr commented:
@kruwed @MarnixCroes

Could you gather all the confusing and wrong messages as a list here (link the issues as well if there are already opened ones).

adamPetho commented on WalletWasabi/WabiSabi/Client/CoinJoinManager.cs:
```diff
@@ -203,6 +203,14 @@ async Task<IEnumerable<SmartCoin>> SanityChecksAndGetCoinCandidatesFunc()
+				// Only unconfirmed coins are available
+				if (!coinCandidates.Any())
+				{
+					throw new CoinJoinClientException(CoinjoinError.NoConfirmedCoinsEligibleToMix, "No confirmed candidates coins available to mix.");
```
Isn't this just too long for a MusicBox message?

How about?
```suggestion
					throw new CoinJoinClientException(CoinjoinError.NoConfirmedCoinsEligibleToMix, "No confirmed coins available to mix.");
```

MaxHillebrand commented:
> We will be aiming to have the fewest messages in the default MusicBox workflow while having as many clear and straightforward messages as we need for the edge cases.

cACK, very good! `The easiest default, with optional insight and control`

Kruwed commented:
> @Kruwed @MarnixCroes
> 
> Could you gather all the confusing and wrong messages as a list here (link the issues as well if there are already opened ones).

-After a coinjoin was failed to be signed by other participants, the message "Awaiting the blame round" briefly appears before switching to "Awaiting other participants".  Although this message is accurate, it is not helpful for users who are trying to figure out why the coinjoin timer has reset again.  The key word here for replacement is **blame**. 

-When a round surpasses the number of allowed inputs and is split, the message "Round completed, awaiting the next round" appears.  This message is misleading because it implies some sort of advancement when the situation is more like a failure (from the perspective of a user watching the timer count down).  The key word here for replacement is **completed**.

-When a user has unconfirmed coins, the message "Insufficient funds eligible for coinjoin" appears.  This message is misleading because it implies there is something wrong with the _amount_ of coins in the wallet rather than coins being in a temporary state that requires waiting.  The key word here for replacement is **Insufficient**.

I could not find open issues besides the one I made for the "Insufficient" message - https://github.com/zkSNACKs/WalletWasabi/issues/10947

soosr commented:
> -After a coinjoin was failed to be signed by other participants, the message "Awaiting the blame round" briefly appears before switching to "Awaiting other participants". Although this message is accurate, it is not helpful for users who are trying to figure out why the coinjoin timer has reset again. The key word here for replacement is **blame**.

Won't be fixed.

> -When a round surpasses the number of allowed inputs and is split, the message "Round completed, awaiting the next round" appears. This message is misleading because it implies some sort of advancement when the situation is more like a failure (from the perspective of a user watching the timer count down). The key word here for replacement is **completed**.

Fixed.

> -When a user has unconfirmed coins, the message "Insufficient funds eligible for coinjoin" appears. This message is misleading because it implies there is something wrong with the _amount_ of coins in the wallet rather than coins being in a temporary state that requires waiting. The key word here for replacement is **Insufficient**.

Fixed.

soosr commented on WalletWasabi.Fluent/ViewModels/Wallets/CoinJoinStateViewModel.cs:
```diff
@@ -34,7 +34,12 @@ public partial class CoinJoinStateViewModel : ViewModelBase
 	private const string GeneralErrorMessage = "Awaiting valid conditions";
+	private const string WaitingForConfirmedFunds = "Awaiting confirmed funds";
+	private const string CoinsRejectedMessage = "Some funds are rejected from coinjoining";
+	private const string OnlyImmatureCoinsAvailableMessage = "Only immature funds are available";
```
This is only for miners, ok to have a message like this.

MaxHillebrand reviewed (approved):
Well done

yahiheb reviewed (commented):
What is the difference between `Awaiting coinjoin` and `Awaiting a round`?

I would replace the word `funds` with `coins` everywhere, and if the word coin is technical for our users to understand then what shall we say about the word coinjoin which is more used in these musicbox messages.

I would also not use the term `round` alone, because if users don't know what coin/coinjoin is then how can they understand what a round is?

nopara73 commented:
> What is the difference between Awaiting coinjoin and Awaiting a round?

Coinjoin is a type of transaction, while round, in this context means: coinjoin round, which denotes a phase where coinjoin happens. Therefore in my understanding awaiting coinjoin and awaiting a round means the same thing.

> I would replace the word funds with coins everywhere, and if the word coin is technical for our users to understand then what shall we say about the word coinjoin which is more used in these musicbox messages.
> I would also not use the term round alone, because if users don't know what coin/coinjoin is then how can they understand what a round is?

Generally ACK, but let's put away the hammers and rather examine each and every sentence on a case-by-case basis and with context. Otherwise, we might end up with awkward wordings, like changing `Round ended, awaiting next round` to `Coinjoin ended, awaiting next coinjoin`, which would be just dumb.

soosr commented:
@zkSNACKs/code-team 
@zkSNACKs/ui-team 

This PR should be also tested by anyone else except me. Please try it out and do some CJs.

molnard reviewed (approved):
tACK

MarnixCroes reviewed (approved):
tested LGTM
few comments/nits, maybe out of scope/for follow up PR

- Could we add another message for when there is no internet connection (so not only that backend is not connected)? So if user starts coinjoin and loses internet,  the musicbox displays a more appropriate message instead of the `CoinsRejectedMessage`.
- If user starts coinjoining and the client is `Aborting input registrations: 'WrongPhase'` the `RoundFinishedMessage` is flashed real quick and then the `CoinsRejectedMessage` is displayed for a couple seconds (nothing incorrect but not great UX imo)
- When the `CoinJoinCoinSelector.SelectCoinsForRound (85)`  finds `No suitable coins for this round.` the `NoCoinsEligibleToMixMessage` is displayed in musicbox, which is a bit confusing as I do have a coin which is eligible.

adamPetho commented on WalletWasabi/WabiSabi/Client/CoinJoinManager.cs:
```diff
@@ -282,6 +276,64 @@ void StopCoinJoinCommand(StopCoinJoinCommand stopCommand)
 		await WaitAndHandleResultOfTasksAsync(nameof(trackedAutoStarts), trackedAutoStarts.Values.Select(x => x.Task).ToArray()).ConfigureAwait(false);
 	}
 
+	private async Task<IEnumerable<SmartCoin>> SelectCandidateCoinsAsync(IWallet walletToStart, int bestHeight)
```
Nit (feel free to ignore):

Previously this method returned a new `CoinsView`. 
Do we want to lose that? 

We can easily wrap the `coinCandidates` in the return. `return new CoinsView(coinCandidates)`
Then we can change the return value of the method.

soosr replied:
> Do we want to lose that?

I don't know. Why was that? What is the benefit?

adamPetho replied:
> What is the benefit?

Now that I looked at the code deeper, I think there are no benefits for my change. Ignore it.

adamPetho reviewed (approved):
LGTM
